sched/wqueue/kwork_cancel.c

```diff
@@ -86,14 +83,15 @@ static int work_qcancel(FAR struct kwork_wqueue_s *wqueue, bool sync,
           if (wqueue->worker[wndx].work == work &&
               wqueue->worker[wndx].pid != nxsched_gettid())
             {
+              wqueue->worker[wndx].wait_count--;
+              spin_unlock_irqrestore(&g_wqueue_lock, flags);
               nxsem_wait_uninterruptible(&wqueue->worker[wndx].wait);
-              ret = 1;
-              break;
+              return 1;
```

Comment:
Returning a positive value means the work queue was removed, but why wasn't it nullified ( wqueue->worker[wndx].work = NULL; ) ?

Reply:
will be set at this line 
https://github.com/apache/nuttx/pull/14623/commits/d7d8355e1e30b0931b26f684b7cd3319c77f88b2#diff-e692bbd69daa1470b1d56a37ed008bd703cf7e0ed3920a0b6d6688673c3c75d2R199

Reply:
Thank you!